examples/demo/updating_plot/updating_plot5.py

```diff
@@ -32,27 +32,32 @@
 
 
 class AnimatedPlot(HasTraits):
-    def __init__(self, x, y, color="blue", bgcolor="none", orientation="h"):
-        self.y_values = y[:]
-        if type(x) == ArrayDataSource:
-            self.x_values = x.get_data()[:]
+    x_values = Any()  # Array or ArraryDataSource
+    y_values = Array()
+    color = Str()
+    orientation = Enum("h", "v")
+
+    plot = Instance(Component)
+
+    def _plot_default(self):
+        if type(self.x_values) == ArrayDataSource:
+            self.x_values = self.x_values.get_data()[:]
             plot = create_line_plot(
-                (x, self.y_values),
-                color=color,
-                bgcolor=bgcolor,
+                (self.x_values, self.y_values),
+                color=self.color,
+                bgcolor="none",
```

Comment:
setting bgcolor to `none` here unlike the other examples where it is set to `white` makes the plot hard to read. on windows, the background was grey/black when it is set to `none`